fvm/utils/test.go

```diff
@@ -0,0 +1,131 @@
+package utils
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type SimpleView struct {
+	Parent *SimpleView
+	Ledger *MapLedger
+}
+
+func NewSimpleView() *SimpleView {
+	return &SimpleView{
+		Ledger: &MapLedger{
+			Registers:       make(map[string]flow.RegisterEntry),
+			RegisterTouches: make(map[string]bool),
+		},
+	}
+}
+
+func (v *SimpleView) NewChild() state.View {
+	ch := NewSimpleView()
+	ch.Parent = v
+	return ch
+}
+
+func (v *SimpleView) MergeView(o state.View) error {
+	var other *SimpleView
+	var ok bool
+	if other, ok = o.(*SimpleView); !ok {
+		return fmt.Errorf("can not merge: view type mismatch (given: %T, expected:SimpleView)", o)
+	}
+
+	for _, item := range other.Ledger.Registers {
+		err := v.Ledger.Set(item.Key.Owner, item.Key.Controller, item.Key.Key, item.Value)
+		if err != nil {
+			return fmt.Errorf("can not merge: %w", err)
+		}
+	}
```

Comment:
Just as a heads up this implementation seems **non-atomic**, i.e., if an error occurs it leaves the merge incomplete on the _original_ ledger. Maybe as a test helper, it is not a critical constraint to be atomic.

Reply:
thats okey, merge view failures cause panics and terminate the nodes.